source/level_set_base.cc

```diff
@@ -63,33 +69,71 @@ LevelSetBaseAlgorithm<dim>::LevelSetBaseAlgorithm(const FlowParameters &paramete
   {
     interpolation_concentration_pressure.reinit(
       this->navier_stokes.get_fe_p().dofs_per_cell, this->fe->dofs_per_cell);
-    const FE_Q_iso_Q1<dim> &fe_mine = dynamic_cast<const FE_Q_iso_Q1<dim> &>(*this->fe);
-    const std::vector<unsigned int> lexicographic_ls =
-      fe_mine.get_poly_space_numbering_inverse();
-    if (const FE_Q<dim> *fe_p =
-          dynamic_cast<const FE_Q<dim> *>(&this->navier_stokes.get_fe_p()))
+    if (this->parameters.use_simplex_mesh)
       {
+        const auto fe_mine = dynamic_cast<const Simplex::FE_P<dim> *>(this->fe.get());
+        const auto fe_p =
+          dynamic_cast<const Simplex::FE_P<dim> *>(&this->navier_stokes.get_fe_p());
+
+        Assert(fe_mine, ExcNotImplemented());
+        Assert(fe_p, ExcNotImplemented());
+
+        /*
+        const std::vector<unsigned int> lexicographic_ls =
+          fe_mine->get_poly_space_numbering_inverse();
+
         const std::vector<unsigned int> lexicographic_p =
           fe_p->get_poly_space_numbering_inverse();
+        */
+
+        std::vector<unsigned int> lexicographic_ls(fe_mine->dofs_per_cell);
+        std::vector<unsigned int> lexicographic_p(fe_p->dofs_per_cell);
+
+        for (unsigned int j = 0; j < fe_mine->dofs_per_cell; ++j)
+          lexicographic_ls[j] = j;
+
+        for (unsigned int j = 0; j < fe_p->dofs_per_cell; ++j)
+          lexicographic_p[j] = j;
```

Comment:
I am not sure about these lines in the context of simplex elements. I simply count up here. How are these permutations relevant?

This is related to https://github.com/dealii/dealii/pull/11353.

Reply:
This seems correct. They are used for the same purpose as you used the renumbering from `shape_info` in a dealii PR I reviewed today regarding the computation of a matrix from `FEEvaluation` - one needs a compatible numbering with the way `MatrixFree`/`FEEvaluation` sorts its indices. I think this function was added before we had the renumbering exposed in shape info, but one could of course now switch to shape info as well.

Reply:
Oh I see. Thanks for the clarification! This is not relevant for simplex+MF at the moment and probably won't be relevant!?

Reply:
It is definitely not relevant with the implementation we have right now. However, if we were adding truncated tensor product evaluation to the simplex elements (which is in `MatrixFree` for `FE_DGP`), we would need a renumbering because `FE_P` enumerates unknowns in a hierarchical way, which is not the truncated lexicographic one.